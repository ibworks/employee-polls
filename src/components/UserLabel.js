const UserLabel = ({ user }) => {
    if (!user) return null;

    const avatar = user.avatarURL.length > 0
        ? (
            <img
                className="avatar-small"
                src={user.avatarURL}
                alt="Avatar" />)
        : null;
    return (
        <div className="user-label">
            { avatar }            
            <div className="user-label-name">
                <strong>{user.name}</strong>
                {user.id}
            </div>
        </div>);
};

export default UserLabel;