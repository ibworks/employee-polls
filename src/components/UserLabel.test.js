import { render } from '@testing-library/react';
import * as React from 'react';
import UserLabel from './UserLabel';

const user = {
    id: "testId",
    name: "test name",
    avatarURL: "https://media.istockphoto.com/vectors/user-avatar-profile-icon-black-vector-illustration-on-transparent-vector-id1313958250?b=1&k=20&m=1313958250&s=170667a&w=0&h=9SnigCTdaqo6v-R24TabGaIczGLG-5m6rkY6cMSDbgM=", 
}

describe('UserLabel', () => {
    it('will match snapshot', () => {
        var component = render(<UserLabel user={user} />);
        expect(component).toMatchSnapshot();
    });
    it('will match snapshot without user', () => {
        var component = render(<UserLabel />);
        expect(component).toMatchSnapshot();
    });

    it('will have expected fields', () => {
        var component = render(<UserLabel user={user} />);
        var result = component.getByText(user.name);
        expect(result).toBeInTheDocument();
    });
});
