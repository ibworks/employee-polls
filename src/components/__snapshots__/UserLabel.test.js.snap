// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLabel will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-label"
      >
        <img
          alt="Avatar"
          class="avatar-small"
          src="https://media.istockphoto.com/vectors/user-avatar-profile-icon-black-vector-illustration-on-transparent-vector-id1313958250?b=1&k=20&m=1313958250&s=170667a&w=0&h=9SnigCTdaqo6v-R24TabGaIczGLG-5m6rkY6cMSDbgM="
        />
        <div
          class="user-label-name"
        >
          <strong>
            test name
          </strong>
          testId
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-label"
    >
      <img
        alt="Avatar"
        class="avatar-small"
        src="https://media.istockphoto.com/vectors/user-avatar-profile-icon-black-vector-illustration-on-transparent-vector-id1313958250?b=1&k=20&m=1313958250&s=170667a&w=0&h=9SnigCTdaqo6v-R24TabGaIczGLG-5m6rkY6cMSDbgM="
      />
      <div
        class="user-label-name"
      >
        <strong>
          test name
        </strong>
        testId
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserLabel will match snapshot without user 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
